src/DBConnection.cpp

```diff
@@ -343,6 +361,23 @@ int DBConnection::FastMode(const char *schema /* = "main" */)
    return ModeConfig(mDB, schema, FastConfig);
 }
 
+int DBConnection::SetPageSize(const char* schema)
+{
+   // First of all - let's check if the database is empty.
+   // Otherwise, VACUUM can take a significant amount of time.
+   // VACUUM is required to force SQLite3 to change the page size.
+   // This function will be the first called on the connection,
+   // so if DB is empty we can assume that journal was not
+   // set to WAL yet.
+   int rc = sqlite3_exec(
+      mDB, "SELECT 1 FROM project LIMIT 1;", nullptr, nullptr, nullptr);
```

Comment:
I haven't seen this technique before... not SELECT * ?

Reply:
`SELECT 1 ...` means that the engine doesn't have to read any data. This is the fastest possible query in this case because we only care that the `project` table has a row in it.